addons/binding/org.openhab.binding.wifiled/src/main/java/org/openhab/binding/wifiled/handler/FadingWiFiLEDDriver.java

```diff
@@ -59,12 +74,29 @@ public void init() throws IOException {
             dtoState = LEDStateDTO.valueOf(s.state, s.program, s.programSpeed, s.red, s.green, s.blue, s.white,
                     s.white2);
             power = (s.state & 0x01) != 0;
-            currentTargetState = InternalLedState.fromRGBW(s.red, s.green, s.blue, s.white, s.white2);
+            currentFaderState = currentState = InternalLedState.fromRGBW(s.red, s.green, s.blue, s.white, s.white2);
         } catch (IOException ignored) {
         }
     }
 
     @Override
+    public void dispose() {
+        waiterFuture.shutdown();
```

Comment:
Martin,

we are actually not disposing the the handler but we are disposing the driver. WiFiLEDHandler will completely remove the driver in dispose and create a new one in WiFiLEDHandler.initialise.
I renamed the dispose method to shutdown for clarification and avoid further confusion.